_posts/en/posts/2016-06-08-version-bits-miners.md

```diff
@@ -0,0 +1,102 @@
+
+## What is the activation workflow?
+
+Under _version bits_, a soft fork proposal goes through a workflow: 
+
+- `[DEFINED]` -> `[STARTED]` -> `[LOCKED_IN]` -> `[ACTIVE]`
+
+or
+
+- `[DEFINED]` -> `[STARTED]` -> `[FAILED]`
+
+![version bits state diagram](https://raw.githubusercontent.com/bitcoin/bips/master/bip-0009/states.png)
+
+The Bitcoin network retargets mining difficulty every 2016 blocks; at this time _version bits_ will look at the window of the previous 2016 every to see how many blocks signal for a given soft fork. If 95% of the blocks signal readiness for the soft fork, the state changes from `[STARTED]` to `[LOCKED_IN]`.
+
+After `[LOCKED_IN]` the rules will activate after one more difficulty retarget, i.e. a further 2016 blocks. Nodes software will warn that an upgrade is pending.
+
+## What is the version bit?
+
+When no soft forks are being signalled, miners should set the block version field to `0x20000000`.
+
+## When should miners set bits?
+
+To signal readiness for soft fork(s), miners should set the relevant version bit(s) together with `0x20000000`. This should only be done after the soft fork's _start time_. 
+
+The bits should be unset if either the soft fork activates, or reached `[FAILED]` state.
+
+For example:
+
+"alice" soft fork uses bit 0, i.e. `0x1` + `0x20000000`
+
+|0|0|1|0|0| ... |0|0|0|0|0|0|0|0|0|1|
+
+"bob" soft fork uses bit, 1, i.e. `0x2` + `0x20000000`
+
+|0|0|1|0|0| ... |0|0|0|0|0|0|0|0|1|0|
+
+To signal both soft forks at once, use `0x20000003` (i.e. `0x1` + `0x2` + `0x20000000`*)
+
+|0|0|1|0|0| ... |0|0|0|0|0|0|0|0|1|1|
+
+* note if once activated before the other, you must unset the relevant bit and continue signalling the other.
+
+## How does it differ to an ISM soft fork?
+
+IsSuperMajority() or ISM for short, is a legacy soft fork trigger that activates new rules once 950 out of 1000 blocks are mined which signal the new block version.
+
+1. An IsSuperMajority() soft fork will orphan all blocks with previous version after activation.
+
+For example, if the current version is 4, and the next soft fork introduces version 5 blocks, then after activation is reached (950/1000 blocks), nodes will reject all version 4 blocks.
+
+Once a _version bits_ soft fork reached activation, nodes will simply begin enforcing the new rules, and will NOT orphan a non-signalling block.
+
+2. ISM() look at the previous 1000 blocks on a rolling basis; _version bits_ looks at the previous 2016 block once each time the mining difficulty retargets.
+
+3. ISM() soft forks do not expire. _version bits_ soft forks can only activate between the _start time_ and the _timeout_.
+
+## Do miners have to upgrade?
+
+No. A [BIP9][] soft fork will not activate unless 95% of the miners signal readiness. If a soft fork reaches `[LOCKED_IN]` state, where the vast majority of the miners are ready for the change, the remaining miners should upgrade within roughly 2 weeks before the next difficulty retarget.
```

Comment:
The time here is ambiguous: it sounds like we're saying they should update (next_retarget - 2 weeks) instead of (next_retarget ~= 2 weeks).  I suggest saying: ...the remaining miners should upgrade before the next difficulty retarget (about 2 weeks).